<?xml version="1.0" encoding="UTF-8"?>
<!--This does not replace URDF, and is not an extension
of URDF.
    This is a format for representing semantic information about the robot structure.
    A URDF file must exist for this robot as well, where the joints and the links that are referenced
are defined
-->
<robot name="g1_dual_arm">
    <!--GROUPS:
    Representation of a set of joints and links. This can be useful for specifying DOF to plan for,
    defining arms, end effectors, etc-->
    <!--LINKS:
    When a link is specified, the parent joint of that link (if it exists) is automatically
    included-->
    <!--JOINTS:
    When a joint is specified, the child link of that joint (which will always exist) is
    automatically included-->
    <!--CHAINS:
    When a chain is specified, all the links along the chain (including endpoints) are included in
    the group. Additionally, all the joints that are parents to included links are also included.
    This means that joints along the chain and the parent joint of the base link are included in the
    group-->
    <!--SUBGROUPS:
    Groups can also be formed by referencing to already defined group names-->
    <group name="g1_left_arm">
        <joint name="fixed_joint" />
        <joint name="waist_roll_joint" />
        <joint name="waist_pitch_joint" />
        <joint name="left_shoulder_pitch_joint" />
        <joint name="left_shoulder_roll_joint" />
        <joint name="left_shoulder_yaw_joint" />
        <joint name="left_elbow_joint" />
        <joint name="left_wrist_roll_joint" />
        <joint name="left_wrist_pitch_joint" />
        <joint name="left_wrist_yaw_joint" />
        <joint name="left_hand_palm_joint" />
    </group>
    <group name="g1_left_hand">
        <link name="left_rubber_hand" />
    </group>
    <group name="g1_right_arm">
        <joint name="fixed_joint" />
        <joint name="waist_roll_joint" />
        <joint name="waist_pitch_joint" />
        <joint name="right_shoulder_pitch_joint" />
        <joint name="right_shoulder_roll_joint" />
        <joint name="right_shoulder_yaw_joint" />
        <joint name="right_elbow_joint" />
        <joint name="right_wrist_roll_joint" />
        <joint name="right_wrist_pitch_joint" />
        <joint name="right_wrist_yaw_joint" />
        <joint name="right_hand_palm_joint" />
    </group>
    <group name="g1_right_hand">
        <link name="right_rubber_hand" />
    </group>
    <!--GROUP
    STATES: Purpose: Define a named state for a particular group, in terms of joint values. This is
    useful to define states like 'folded arms'-->
    <group_state name="g1_left_arm_home" group="g1_left_arm">
        <joint name="left_elbow_joint" value="1.5707" />
        <joint name="left_shoulder_pitch_joint" value="0" />
        <joint name="left_shoulder_roll_joint" value="0" />
        <joint name="left_shoulder_yaw_joint" value="0" />
        <joint name="left_wrist_pitch_joint" value="0" />
        <joint name="left_wrist_roll_joint" value="0" />
        <joint name="left_wrist_yaw_joint" value="0" />
    </group_state>
    <group_state name="g1_right_arm_home" group="g1_right_arm">
        <joint name="right_elbow_joint" value="1.5707" />
        <joint name="right_shoulder_pitch_joint" value="0" />
        <joint name="right_shoulder_roll_joint" value="0" />
        <joint name="right_shoulder_yaw_joint" value="0" />
        <joint name="right_wrist_pitch_joint" value="0" />
        <joint name="right_wrist_roll_joint" value="0" />
        <joint name="right_wrist_yaw_joint" value="0" />
    </group_state>
    <!--END
    EFFECTOR: Purpose: Represent information about an end effector.-->
    <end_effector name="g1_left_end_effector" parent_link="left_wrist_yaw_link" group="g1_left_hand"
        parent_group="g1_left_arm" />
    <end_effector name="g1_right_end_effector" parent_link="right_wrist_yaw_link"
        group="g1_right_hand" parent_group="g1_right_arm" />
    <virtual_joint name="fixed_joint" type="fixed" parent_frame="world" child_link="waist_yaw_link" />
    <!--DISABLE
    COLLISIONS: By default it is assumed that any link of the robot could potentially come into
    collision with any other link in the robot. This tag disables collision checking between a
    specified pair of links. -->
    <disable_collisions link1="head_link" link2="left_shoulder_pitch_link" reason="Never" />
    <disable_collisions link1="head_link" link2="logo_link" reason="Never" />
    <disable_collisions link1="head_link" link2="right_shoulder_pitch_link" reason="Never" />
    <disable_collisions link1="head_link" link2="torso_link" reason="Adjacent" />
    <disable_collisions link1="head_link" link2="waist_support_link" reason="Never" />
    <disable_collisions link1="left_elbow_link" link2="left_shoulder_pitch_link" reason="Never" />
    <disable_collisions link1="left_elbow_link" link2="left_shoulder_roll_link" reason="Never" />
    <disable_collisions link1="left_elbow_link" link2="left_shoulder_yaw_link" reason="Adjacent" />
    <disable_collisions link1="left_elbow_link" link2="left_wrist_pitch_link" reason="Never" />
    <disable_collisions link1="left_elbow_link" link2="left_wrist_roll_link" reason="Adjacent" />
    <disable_collisions link1="left_elbow_link" link2="left_wrist_yaw_link" reason="Never" />
    <disable_collisions link1="left_shoulder_pitch_link" link2="left_shoulder_roll_link"
        reason="Adjacent" />
    <disable_collisions link1="left_shoulder_pitch_link" link2="left_shoulder_yaw_link"
        reason="Never" />
    <disable_collisions link1="left_shoulder_pitch_link" link2="left_wrist_pitch_link"
        reason="Never" />
    <disable_collisions link1="left_shoulder_pitch_link" link2="left_wrist_roll_link" reason="Never" />
    <disable_collisions link1="left_shoulder_pitch_link" link2="logo_link" reason="Never" />
    <disable_collisions link1="left_shoulder_pitch_link" link2="right_shoulder_pitch_link"
        reason="Never" />
    <disable_collisions link1="left_shoulder_pitch_link" link2="right_shoulder_roll_link"
        reason="Never" />
    <disable_collisions link1="left_shoulder_pitch_link" link2="right_shoulder_yaw_link"
        reason="Never" />
    <disable_collisions link1="left_shoulder_pitch_link" link2="torso_link" reason="Adjacent" />
    <disable_collisions link1="left_shoulder_pitch_link" link2="waist_support_link" reason="Never" />
    <disable_collisions link1="left_shoulder_roll_link" link2="left_shoulder_yaw_link"
        reason="Adjacent" />
    <disable_collisions link1="left_shoulder_roll_link" link2="left_wrist_roll_link" reason="Never" />
    <disable_collisions link1="left_shoulder_roll_link" link2="logo_link" reason="Never" />
    <disable_collisions link1="left_shoulder_roll_link" link2="right_shoulder_pitch_link"
        reason="Never" />
    <disable_collisions link1="left_shoulder_roll_link" link2="right_shoulder_roll_link"
        reason="Never" />
    <disable_collisions link1="left_shoulder_roll_link" link2="waist_support_link" reason="Never" />
    <disable_collisions link1="left_shoulder_yaw_link" link2="left_wrist_pitch_link" reason="Never" />
    <disable_collisions link1="left_shoulder_yaw_link" link2="left_wrist_roll_link" reason="Never" />
    <disable_collisions link1="left_shoulder_yaw_link" link2="left_wrist_yaw_link" reason="Never" />
    <disable_collisions link1="left_shoulder_yaw_link" link2="right_shoulder_pitch_link"
        reason="Never" />
    <disable_collisions link1="left_shoulder_yaw_link" link2="waist_support_link" reason="Never" />
    <disable_collisions link1="left_wrist_pitch_link" link2="left_wrist_roll_link" reason="Adjacent" />
    <disable_collisions link1="left_wrist_pitch_link" link2="left_wrist_yaw_link" reason="Adjacent" />
    <disable_collisions link1="left_wrist_roll_link" link2="left_wrist_yaw_link" reason="Never" />
    <disable_collisions link1="left_wrist_yaw_link" link2="left_rubber_hand" reason="Never" />
    <disable_collisions link1="logo_link" link2="right_shoulder_pitch_link" reason="Never" />
    <disable_collisions link1="logo_link" link2="right_shoulder_roll_link" reason="Never" />
    <disable_collisions link1="logo_link" link2="torso_link" reason="Adjacent" />
    <disable_collisions link1="logo_link" link2="waist_support_link" reason="Never" />
    <disable_collisions link1="right_elbow_link" link2="right_shoulder_pitch_link" reason="Never" />
    <disable_collisions link1="right_elbow_link" link2="right_shoulder_roll_link" reason="Never" />
    <disable_collisions link1="right_elbow_link" link2="right_shoulder_yaw_link" reason="Adjacent" />
    <disable_collisions link1="right_elbow_link" link2="right_wrist_pitch_link" reason="Never" />
    <disable_collisions link1="right_elbow_link" link2="right_wrist_roll_link" reason="Adjacent" />
    <disable_collisions link1="right_elbow_link" link2="right_wrist_yaw_link" reason="Never" />
    <disable_collisions link1="right_shoulder_pitch_link" link2="right_shoulder_roll_link"
        reason="Adjacent" />
    <disable_collisions link1="right_shoulder_pitch_link" link2="right_shoulder_yaw_link"
        reason="Never" />
    <disable_collisions link1="right_shoulder_pitch_link" link2="right_wrist_pitch_link"
        reason="Never" />
    <disable_collisions link1="right_shoulder_pitch_link" link2="right_wrist_roll_link"
        reason="Never" />
    <disable_collisions link1="right_shoulder_pitch_link" link2="torso_link" reason="Adjacent" />
    <disable_collisions link1="right_shoulder_pitch_link" link2="waist_support_link" reason="Never" />
    <disable_collisions link1="right_shoulder_roll_link" link2="right_shoulder_yaw_link"
        reason="Adjacent" />
    <disable_collisions link1="right_shoulder_roll_link" link2="right_wrist_roll_link"
        reason="Never" />
    <disable_collisions link1="right_shoulder_roll_link" link2="waist_support_link" reason="Never" />
    <disable_collisions link1="right_shoulder_yaw_link" link2="right_wrist_pitch_link"
        reason="Never" />
    <disable_collisions link1="right_shoulder_yaw_link" link2="right_wrist_roll_link" reason="Never" />
    <disable_collisions link1="right_shoulder_yaw_link" link2="right_wrist_yaw_link" reason="Never" />
    <disable_collisions link1="right_shoulder_yaw_link" link2="waist_support_link" reason="Never" />
    <disable_collisions link1="right_wrist_pitch_link" link2="right_wrist_roll_link"
        reason="Adjacent" />
    <disable_collisions link1="right_wrist_pitch_link" link2="right_wrist_yaw_link"
        reason="Adjacent" />
    <disable_collisions link1="right_wrist_roll_link" link2="right_wrist_yaw_link" reason="Never" />
    <disable_collisions link1="right_wrist_yaw_link" link2="right_rubber_hand" reason="Never" />
    <disable_collisions link1="torso_link" link2="waist_support_link" reason="Adjacent" />
</robot>
